dspace-api/src/main/java/org/dspace/statistics/SolrLoggerServiceImpl.java

```diff
@@ -834,9 +874,17 @@ public void process(List<SolrInputDocument> docs)
                     }
                 }
             }
-            solr.add(solrDocument);
+
+            solrDocument.removeField("_version_");
+            solrDocument.removeField("[shard]");
+
+            shard.add(solrDocument);
```

Comment:
Could you add some inline comments regarding the shard logic here?  It's not clear at a glance why we are removing the version and shard fields, and then adding the entire solrDocument back to the shard.